profile-specific-challenges/frontend/components/idioms.jsx

```diff
@@ -319,60 +397,49 @@ export function UnnecessaryFunctionRedefinitions(emails) {
   );
 }
 
-async function fetchRecords() {
-  return [{ id: 1, type: "record" }];
-}
-async function fetchAlternateRecords() {
-  return [{ id: 1, type: "alt-record" }];
-}
-
 export function SerialLoading() {
   const [records, setRecords] = useState([]);
   const [altRecords, setAltRecords] = useState([]);
 
   useEffect(() => {
     async function loadRecords() {
-      const recs = await fetchRecords();
-      const altRecs = await fetchAlternateRecords();
+      const [recs, altRecs] = await Promise.all([
+        API.fetchRecords(),
+        API.fetchAlternateRecords(),
+      ]);
+
       setRecords(recs);
       setAltRecords(altRecs);
     }
     loadRecords();
   }, []);
 
   return (
-    <div>
+    <ul>
       {records.map((rec) => (
-        <div key={rec.id}></div>
+        <li key={rec.id}>{rec.name}</li>
       ))}
       {altRecords.map((rec) => (
-        <div key={rec.id}></div>
+        <li key={rec.id}>{rec.name}</li>
       ))}
-    </div>
+    </ul>
   );
 }
 
-async function fetchRecords() {
-  return [{ id: 1, type: "record" }];
-}
-async function fetchAlternateRecords() {
-  return [{ id: 1, type: "alt-record" }];
-}
-
 // Hint: part of the rendering structure is re-rendered frequently unnecessarily
-export function UnoptimizableRenderingStructure(altRecords) {
+export function UnoptimizableRenderingStructure({ altRecords }) {
```

Comment:
Not sure why this is called UnoptimizableRenderingStructure, what about the rendered HTML is wrong? Maybe keys colliding? As for the use effect it was wrong, but I'm assuming we want to fetch records every 5 seconds anyways.